From b580a8256f87b8fdf3406eeac465a7d1f82235c2 Mon Sep 17 00:00:00 2001
From: Marina Polyakova <m.polyakova@postgrespro.ru>
Date: Wed, 24 Jan 2024 10:15:55 +0300
Subject: [PATCH] Retire MemoryContextResetAndDeleteChildren() macro.

Caused by the following commits in PostgreSQL:
- eaa5808e8ec4e82ce1a87103a6b6f687666e4e4c (PostgreSQL 9.5)
  Redefine MemoryContextReset() as deleting, not resetting, child contexts.
- 6a72c42fd5af7ada49584694f543eb06dddb4a87 (PostgreSQL 17)
  Retire MemoryContextResetAndDeleteChildren() macro.
---
 src/shared_ispell.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shared_ispell.c b/src/shared_ispell.c
index 4fac4fb..37243e2 100644
--- a/src/shared_ispell.c
+++ b/src/shared_ispell.c
@@ -633,7 +633,7 @@ dispell_lexize(PG_FUNCTION_ARGS)
 		 * info here
 		 */
 
-		MemoryContextResetAndDeleteChildren(saveInfo.infoCntx);
+		MemoryContextReset(saveInfo.infoCntx);
 		MemSet(info, 0, sizeof(*info));
 
 		init_shared_dict(info, saveInfo.infoCntx, saveInfo.dictFile,
